astropy/wcs/wcsapi/high_level_api.py

```diff
@@ -91,6 +90,10 @@ def world_to_array_index(self, *world_objects):
         pixel indexing and ordering conventions. The indices should be returned
         as rounded integers.
         """
+        if self.pixel_n_dim == 1:
+            return np.round(self.world_to_pixel(*world_objects)).astype(int)
+        else:
+            return tuple(np.round(self.world_to_pixel(*world_objects)[::-1]).astype(int).tolist())
```

Comment:
I'd prefer to use something like  https://github.com/spacetelescope/gwcs/blob/master/gwcs/utils.py#L52 instead of `round`, to get consistent results of converting to the center of the pixel.

Reply:
I see what I suggested is implemented in the high-level class. What is the reason for these to be different?